CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
 
-Note: An alternative approach that was considered was to augment the probablySupportsContext() API by making it check the second argument.  That approach is difficult to consolidate with how dictionary arguments are meant to work, where unsupported entries are just ignored.
+| colorType | alpha | Implied pixelFormat |
+| - | - | - |
+| "uint8" | false | RGB8 (default) |
+| "uint8" | false | SRGB8 |
+| "uint8" | true | RGBA8 (default) |
+| "uint8" | true | SRGB8_ALPHA8 |
+| "float16" | true | RGBA16F |
 
-#### ImageBitmap
+Note that for pixelFormat of SRGB8 or SRGB8_ALPHA8, the values written to the framebuffer through rasterization operations (e.g, gl_FragColor and blending results) will have the linear-to-sRGB transfer function applied to them. This means than a gl_FragColor value of 0.5 represents half of the intensity of a gl_FragColor of 1.0, and so it is encoded as 0xBC, not 0x80.
 
-TODO(ccameron): Review this
-ImageBitmap objects are augmented to have an internal color space attribute of type CanvasColorSpace and an internal pixelFormat attribute of type CanvasPixelFormat. The colorSpaceConversion creation attribute also accepts enum values that correspond to CanvasColorSpace values. Specifying a CanvasColorSpace value results in a conversion of the image to the specified color space.
+TODO: Should we allow inferring CanvasColorType from WebGL pixel format? Should we simply eliminate CanvasColorType because it is redundant?
 
-#### ImageData
+TODO: Add values for UNPACK_COLORSPACE_CONVERSION_WEBGL to convert to all possible CanvasColorSpace values (e.g, UNPACK_COLORSPACE_CONVERSION_SRGB, UNPACK_COLORSPACE_CONVERSION_P3, UNPACK_COLORSPACE_CONVERSION_REC2020).
 
-TODO(ccameron): Review this
+### ImageData API Changes
 
-IDL
+IDL Additions:
 <pre>
+typedef (Uint8ClampedArray or Uint16Array or Float32Array) ImageDataArray;
 
-enum ImageDataStorageType {
+enum ImageDataStorageFormat {
   "uint8", // default
   "uint16",
   "float32",
 };
 
-typedef (Uint8ClampedArray or Uint16Array or Float32Array) ImageDataArray;
-
-[Constructor(unsigned long sw, unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Constructor(ImageDataArray data, unsigned long sw, optional unsigned long sh, optional ImageDataColorSettings imageDataColorSettings),
- Exposed=(Window,Worker)]
-interface ImageData {
-  readonly attribute unsigned long width;
-  readonly attribute unsigned long height;
-  readonly attribute ImageDataArray data;
+dictionary ImageDataAttributes {
+  CanvasColorSpace colorSpace = "srgb";
+  ImageDataStorageFormat storageFormat = "uint8";
+};
 
-  ImageDataColorSettings getColorSettings();
+partial interface ImageData {
+  ImageDataAttributes getAttributes();
```

Comment:
Why not use a readonly attribute?

Reply:
(because I didn't know it was an option :)